File: src/markov/Graph.java
package markov;

import java.util.List;

public interface Graph<E> {
    boolean add(E node);


    boolean addEdge(E from, E to);

    boolean hasEdge(E from, E to);

    List<E> getNodes();

    List<E> getNeighbors(E node);

    boolean contains(E node);

    int size();

}

File: src/markov_kt/Graph.kt
package markov_kt

interface Graph<E> {
    fun add(node: E): Boolean
    fun addEdge(from: E, to: E): Boolean
    fun hasEdge(from: E, to: E): Boolean
    val nodes: List<E>?
    fun getNeighbors(node: E): List<E>?
    operator fun contains(node: E): Boolean
    fun size(): Int
}